src/m365/spo/commands/tenant/tenant-commandset-add.ts

```diff
@@ -0,0 +1,249 @@
+import { Logger } from '../../../../cli/Logger';
+import GlobalOptions from '../../../../GlobalOptions';
+import { validation } from '../../../../utils/validation';
+import SpoCommand from '../../../base/SpoCommand';
+import commands from '../../commands';
+import Command from '../../../../Command';
+import { Cli, CommandOutput } from '../../../../cli/Cli';
+import { Options as spoListItemAddCommandOptions } from '../listitem/listitem-add';
+import { Options as spoListItemListCommandOptions } from '../listitem/listitem-list';
+import * as spoTenantAppCatalogUrlGetCommand from './tenant-appcatalogurl-get';
+import * as spoListItemAddCommand from '../listitem/listitem-add';
+import * as spoListItemListCommand from '../listitem/listitem-list';
+import { urlUtil } from '../../../../utils/urlUtil';
+import { Solution } from './Solution';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  title: string;
+  listType: string;
+  clientSideComponentId: string;
+  clientSideComponentProperties?: string;
+  webTemplate?: string;
+  location?: string;
+}
+
+class SpoTenantCommandSetAddCommand extends SpoCommand {
+  private static readonly listTypes: string[] = ['List', 'Library', 'SitePages'];
+  private static readonly locations: string[] = ['ContextMenu', 'CommandBar', 'Both'];
+
+  public get name(): string {
+    return commands.TENANT_COMMANDSET_ADD;
+  }
+
+  public get description(): string {
+    return 'Add a ListView Command Set as a tenant wide extension.';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        listType: args.options.listType,
```

Comment:
seems like `listType` is not an optional option

Reply:
We're logging the value here. As it's not PII-sensitive, that's an option. Even required values are interesting to know sometimes .

Reply:
That's true but I guess usually we don't do that 🤔. Or am I wrong 😉

Reply:
If not we should start to 😀

Reply:
fair enough 👍